### ISSUE TITLE
CC on Blackwell

### ISSUE THREAD

**Issue by YurkoWasHere** [NVIDIA/nvtrust#70]
Is there any support for CC on Blackwell GPUs yet? 

If there is would it be part of this repo?

**Comment by Tan-YiFan** [comment]
CC would be enabled on Blackwell. You can refer to https://www.nvidia.com/en-us/data-center/technologies/blackwell-architecture/

**Comment by YurkoWasHere** [comment]
I guess the better question would be would bounce buffer be supported on blackwell,

as i understand it the new  pcie v6 spec of encrypting pcie communication wont be ready for some time.

**Comment by Tan-YiFan** [comment]
Blackwell would support TEE I/O, allowing the GPU to directly read data from CVM private memory. Bounce buffer on CVM is not used during data transfer.

**Comment by YurkoWasHere** [comment]
AFAIK TEE I/O is next gen pcie and probably wont ship for at least a year after Blackwell drops. (I think Intel called it TDX Connect, not sure what the AMD branding is)

If this is incorrect i would love to learn more about possible time lines, otherwise i fear the only way to be early CC on Blackwell is a bounce buffer.

**Comment by Tan-YiFan** [comment]
Yes, it is called TDX connect in Intel TDX.

TDX connect (or TEE I/O) is based on PCIE TDISP (a feature in PCIE 5.0). Could you share with me what feature of TEE I/O is in next gen PCIE?

https://www.nvidia.com/en-us/data-center/technologies/blackwell-architecture/ says blackwell supports TDX I/O.

**Comment by Tan-YiFan** [comment]
The latest version of nvidia open gpu kernel has some updates on Blackwell CC: https://github.com/NVIDIA/open-gpu-kernel-modules/blob/565.57.01/src/nvidia/src/kernel/gpu/conf_compute/arch/blackwell/conf_compute_gb100.c

@rnertney @steven-bellock Could you provide further information?

**Comment by YurkoWasHere** [comment]
Bounce buffer is currently working on Blackwell,

TDIS feels like a long away away as there is no support for it in software yet anywhere i can see.

**Comment by steven-bellock** [comment]
Yes, there's a lot of work that needs to be done for TDISP, especially in the Linux kernel. So it will still be bounce buffers for a while.

**Comment by steven-bellock** [comment]
> The latest version of nvidia open gpu kernel has some updates on Blackwell CC: https://github.com/NVIDIA/open-gpu-kernel-modules/blob/565.57.01/src/nvidia/src/kernel/gpu/conf_compute/arch/blackwell/conf_compute_gb100.c
> 
> [@rnertney](https://github.com/rnertney) [@steven-bellock](https://github.com/steven-bellock) Could you provide further information?

EA / beta support for Blackwell SPT with bounce buffers has been released in `r575` : https://docs.nvidia.com/575trd1-trusted-computing-solutions-release-notes.pdf

**Comment by blossomin** [comment]
> Yes, it is called TDX connect in Intel TDX.
> 
> TDX connect (or TEE I/O) is based on PCIE TDISP (a feature in PCIE 5.0). Could you share with me what feature of TEE I/O is in next gen PCIE?
> 
> https://www.nvidia.com/en-us/data-center/technologies/blackwell-architecture/ says blackwell supports TDX I/O.

Hi yifan @Tan-YiFan , do you know whether this TDISP works with Intel SPR + Hopper 20?

**Comment by Tan-YiFan** [comment]
> > Yes, it is called TDX connect in Intel TDX.
> > TDX connect (or TEE I/O) is based on PCIE TDISP (a feature in PCIE 5.0). Could you share with me what feature of TEE I/O is in next gen PCIE?
> > https://www.nvidia.com/en-us/data-center/technologies/blackwell-architecture/ says blackwell supports TDX I/O.
> 
> Hi yifan [@Tan-YiFan](https://github.com/Tan-YiFan) , do you know whether this TDISP works with Intel SPR + Hopper 20?

H20 doesn't support TDISP, I'm afraid.

**Comment by YurkoWasHere** [comment]
The required hardware was only implemented in Blackwell. However AFAIK there is no software support for TDISP yet.
(Kernel, Drivers, Firmware etc)

**Comment by amshinde** [comment]
@YurkoWasHere I believe this was resolved. Closing this, feel free to reopen, if further assistance or discussion needed.
